component/attr_cache/cacheMap.go

```diff
@@ -55,39 +58,108 @@ func newAttrCacheItem(attr *internal.ObjAttr, exists bool, cachedAt time.Time) *
 		attrFlag: 0,
 		cachedAt: cachedAt,
 	}
-
 	item.attrFlag.Set(AttrFlagValid)
 	if exists {
 		item.attrFlag.Set(AttrFlagExists)
 	}
-
 	return item
 }
 
+func (value *attrCacheItem) insert(attr *internal.ObjAttr, exists bool, cachedAt time.Time) *attrCacheItem {
+	var path string
+	if (attr.Path != "" && attr.Name == "") || attr != nil {
+		path = attr.Path // home/user/folder/file
+		path = internal.TruncateDirName(path)
+	}
+	var itemPath string
+	//start recursion
+	cachedItem := value.insertHelper(attr, exists, cachedAt, path, itemPath)
+	return cachedItem
+}
+
+// TODO: write unit tests for this
+func (value *attrCacheItem) insertHelper(attr *internal.ObjAttr, exists bool, cachedAt time.Time, path string, itemPath string) *attrCacheItem {
+	var cachedItem *attrCacheItem
+	paths := strings.SplitN(path, "/", 2) // paths[0] is home paths[1] is user/folder/file
+	if value.children == nil {
+		value.children = make(map[string]*attrCacheItem)
+	}
+	if len(paths) < 2 {
+		// this is a leaf
+		// we end up with string key being a single folder name instead of a full path. This also will take care of using the folder attribute data.
+		cachedItem = newAttrCacheItem(attr, exists, cachedAt)
+		value.children[paths[0]] = cachedItem
+	} else {
+		itemPath += paths[0] + "/"
+		//see if the directory exists. if not, create it.
+		_, ok := value.children[paths[0]]
+		if !ok {
+			value.children[paths[0]] = newAttrCacheItem(internal.CreateObjAttrDir(itemPath), exists, cachedAt)
+		}
+		cachedItem = value.children[paths[0]].insertHelper(attr, exists, cachedAt, paths[1], itemPath)
+	}
+	return cachedItem
+}
+
+// input: full path to item or file as string
+// output: the attrCacheItem value for the key found in path
+// description: a lookup of any attrCacheItem based on any given full path.
+// TODO: write tests
+func (value *attrCacheItem) get(path string) (*attrCacheItem, error) {
+	path = internal.TruncateDirName(path)
+	paths := strings.Split(path, "/")
+	var currentItem *attrCacheItem
+	var ok bool
```

Comment:
`ok` doesn't need this broader scope

Reply:
as soon as I initialize ok with the `currentItem, ok :=` it provides the error "declared and not used" with the currentItem while ok remains ok. 

The syntax and linting of golang forces me to put these variables declared before the for loop.